### DISCUSSION TITLE
Einige Tags kommen als Error zurück

### DISCUSSION THREAD

**Discussion by SaschaHenning** [spali/go-rscp#13] [Q&A] [answered]
Erstmal Danke für die super Lösung. Ist echt praktisch.

Ich habe leider bei manchen Tags Probleme mit der Abfrage und erkenne nicht direkt, was ich falsch mache:

e3dc.json:
```json
[
"INFO_REQ_TIME",
"EMS_REQ_BAT_SOC",
"EMS_REQ_POWER_ADD",
"EMS_REQ_POWER_BAT",
"EMS_REQ_POWER_HOME",
"EMS_REQ_POWER_GRID",
"EMS_REQ_SELF_CONSUMPTION",
"EMS_REQ_AUTARKY",
"EMS_REQ_GET_POWER_SETTINGS",
"EMS_REQ_GET_MANUAL_CHARGE",
"EMS_REQ_EMERGENCY_POWER_STATUS",
"EP_REQ_IS_GRID_CONNECTED",
"EP_REQ_IS_POSSIBLE",
"EP_REQ_IS_READY_FOR_SWITCH",
"EP_REQ_IS_INVALID_STATE",
"WB_REQ_ENERGY_ALL",
"EMS_REQ_STATUS",
"BAT_REQ_INFO"
]
```

Ergebnis (Auszug):
```
result: 
EMS_POWER_BAT: -1266
EMS_POWER_HOME: 1167
EMS_AUTARKY: 99.83789
...
BAT_INFO: ERR_FORMAT
INFO_TIME: '2022-01-31T22:33:58.000416Z'
EP_IS_READY_FOR_SWITCH: true
EP_IS_GRID_CONNECTED: true
EP_IS_INVALID_STATE: false
EP_IS_POSSIBLE: true
WB_ENERGY_ALL: ERR_FORMAT
```

Warum bekommen `BAT_INFO` und `WB_ENERGY_ALL` als Ergebnis `ERR_FORMAT`?
Wie kann ich das "debuggen"?

Danke für einen Hinweis.

**Comment by spali** [comment, marked as answer]
Ich habe leider keine WB and der E3DC, kann dir darum nur theoretisch helfen.

`WB_ENERGY_ALL` verlangt gemäss Doku von E3DC, dass es innerhalb von REQ_WB_DATA Container verwendet wird. 
siehe `Kann nur innerhalb eines REQ_WB_DATA Container verwendet werden!`

Das heisst, du musst die Tags schachteln und noch ein Index der WB mit liefern.
Ohne es getestet zu haben, müsste es vermutlich in etwa so aussehen:
```json
[
  "EMS_POWER_BAT",
  //....
  [
    "REQ_WB_DATA",
    ["WB_INDEX", 0],
    "WB_REQ_ENERGY_ALL"
  ]
]
```

**Reply by SaschaHenning** [reply]
Danke, habe das übersehen. Kaum macht man es richtig, funktioniert es auch ;-)

**Comment by maikheinrich** [comment]
Genau so verhält es sich mit dem Batterie Tag. Du must eine verschachtelte Anfrage mittels "REQ_BAT_INFO" durchführen und bekommst dann den TAG "BAT_INFO" zurück.

z. B. so...
```
[
  "INFO_REQ_UTC_TIME",
  "INFO_REQ_SW_RELEASE",
  "EMS_REQ_BAT_SOC",
  [
    "BAT_REQ_DATA",
    [
      ["BAT_INDEX", 0],
      "BAT_REQ_DEVICE_NAME",
      "BAT_REQ_DEVICE_STATE",
      "BAT_REQ_MIN_DCB_CELL_TEMPERATURE",
      "BAT_REQ_MAX_DCB_CELL_TEMPERATURE",
      "BAT_REQ_MODULE_VOLTAGE",
      "BAT_REQ_DCB_COUNT",
      "BAT_REQ_CHARGE_CYCLES"

    ]
  ],
  [
    "BAT_REQ_DATA",
    [
      ["BAT_INDEX", 1],
      "BAT_REQ_DEVICE_NAME",
      "BAT_REQ_DEVICE_STATE",
      "BAT_REQ_MIN_DCB_CELL_TEMPERATURE",
      "BAT_REQ_MAX_DCB_CELL_TEMPERATURE",
      "BAT_REQ_MODULE_VOLTAGE",
      "BAT_REQ_DCB_COUNT",
      "BAT_REQ_CHARGE_CYCLES"

    ]
  ]
]
```
LG Maik
